### PR TITLE
PRO-216: Add GetTransactionsByCommitmentID

### PR DESCRIPTION
**Pull request by lithp**
[PRO-216](https://linear.app/worldcoin/issue/PRO-216/merge-getbycommitmentid-into-a-single-gettransactionbycommitmentid)

### CODE REVIEW

File: storage/transaction.go [hunk 1]
```diff
@@ -47,6 +47,26 @@ func (s *Storage) unsafeGetTransactionWithBatchDetails(hash common.Hash) (
...
+	}
+
+	var txs models.GenericTransactionArray
+
+	txs = models.MakeGenericArray()
+	for i := range batchedTxs {
+		txs = txs.AppendOne(batchedTxs[i].ToGenericTransaction())
+	}
+
+	return txs, nil
```
Comment (b-tarczynski) [inline storage/transaction.go:67]: What do you think about below change? It's simpler and we create slice with target capacity.
```suggestion
	txs := make(models.GenericArray, 0, len(batchedTxs))
	for i := range batchedTxs {
		txs = append(txs, batchedTxs[i].ToGenericTransaction())
	}
	return txs, nil
```

File: storage/transaction_test.go [hunk 2]
```diff
@@ -152,6 +152,31 @@ func (s *TransactionTestSuite) TestReplaceFailedTransaction_DoesNotUpdatePending
...
+	otherCommitmentID.IndexInBatch += 1
+	transfer2 := create2Transfer
+	transfer2.Hash = utils.RandomHash()
+	transfer2.CommitmentID = &otherCommitmentID
+	err = s.storage.AddTransaction(&transfer2)
+	s.NoError(err)
+
+	transfers, err := s.storage.GetTransactionsByCommitmentID(txCommitment.ID)
+	s.NoError(err)
+	s.Len(transfers, 1)
```
Comment (b-tarczynski) [inline storage/transaction_test.go:171]: I would add additional check that this transaction equals `transfer1`.

**Review by b-tarczynski — CHANGES_REQUESTED** [review]

File: storage/transaction.go (hunk 1 above)
Comment (lithp) [reply storage/transaction.go]: Huh, I thought I tried this and got a type error, but trying it again now this works, nice catch!

**Comment by lithp** [comment]
Addressed both comments and rebased onto master

**Review by b-tarczynski — APPROVED** [review]

**Pull request merged**